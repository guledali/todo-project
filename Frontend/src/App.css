.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
}

.App-header {
  /* background-color: #282c34; */
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.screen {
  height: 100vh;
  width: 100vw;
  background: #f8f9fa;
}

.h-10 {
  height: 10%;
}
.pt-10 {
  padding-top: 5rem;
}

.custom-size {
  font-size: 2rem;
  font-weight: medium;
}
.m25 {
  margin-top: 100%;
}