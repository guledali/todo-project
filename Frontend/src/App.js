import React, { Component } from 'react';
import { request } from 'graphql-request'
// import logo from './logo.svg';
import Tasks from './components/Tasks';
import './App.css';
import Form from './components/Form';


// {this.state.loading && (
//   <p>Loading...</p>
// )}

class App extends Component {
  state = {
    data: {
      "tasks": [
        {
          "id": "cjp43njwfybxt0a03dsjcba8w",
          "task": "drinking"
        }
      ]
    },
    loading: true
  }
  componentDidMount() {
    const query = `
      query LIST_ITEMS {
        tasks {
          id
          task
        }
      }
    `
    
    request('https://my-app-nkefnbuovn.now.sh', query).then(data =>
      this.setState({
        data: data,
        loading: false
      })
    )
  }
  render() {
    console.log(this.state)
    return (
      <div className="screen" data-testid="app-test">
        <nav className="bg-primary h-10 shadow">
          <div className="pb-3">
            <h1 className="display-2 text-center pt-10 font-weight-normal text-primary" >Todo</h1>
          </div>
          <Form />
          <div className="mt-5">
            {this.state.data.tasks.map((task) =>
              <Tasks key={task.id}
                task={task} />
            )}
          </div>
        </nav>
      </div>
    );
  }
}

export default App;
