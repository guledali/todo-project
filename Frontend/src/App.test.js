// import React from 'react';
// import ReactDOM from 'react-dom';
// import App from './App';

// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });


import React from "react";
import { render } from 'react-testing-library'

import App from './App'
 it('snappshot testing', () => {
   const component = render( <App /> ).getByTestId('app-test');
    expect(component).toMatchSnapshot();
 });