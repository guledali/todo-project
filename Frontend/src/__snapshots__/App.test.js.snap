// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snappshot testing 1`] = `
<div
  class="screen"
  data-testid="app-test"
>
  <nav
    class="bg-primary h-10 shadow"
  >
    <div
      class="pb-3"
    >
      <h1
        class="display-2 text-center pt-10 font-weight-normal text-primary"
      >
        Todo
      </h1>
    </div>
    <form
      class="mt-2"
    >
      <div
        class="form-row align-items-center"
      >
        <div
          class="col-5 offset-3"
        >
          <p
            class="custom-size text-dark"
          >
            Task
          </p>
          <input
            class="form-control"
            data-testid="inputForm"
            id="inlineFormInput"
            placeholder="Add your task"
            type="text"
          />
        </div>
        <div
          class="col-auto"
        >
          <button
            class="btn btn-primary btn-lg m25 w-100"
            id="submitButton"
            type="submit"
          >
            Submit
          </button>
        </div>
      </div>
    </form>
    <div
      class="mt-5"
    >
      <div
        class="w-50 mx-auto"
      >
        <ul
          class="list-group"
        >
          <li
            class="list-group-item d-flex justify-content-between align-items-center py-3 px-4 rounded-0"
          >
            <span
              class="lead font-weight-normal"
            >
              drinking
            </span>
            <button
              aria-label="Close"
              class="close"
            >
              <span
                aria-hidden="true"
                class="text-dark font-weight-bold"
                id="deleteTask"
              >
                Ã—
              </span>
            </button>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
